File: Loenn/entities/finalCore.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local finalCore = {}

finalCore.name = "MaggyHelper/FinalCore"
finalCore.depth = -100
finalCore.fieldInformation = {
    health = {
        minimumValue = 1
    },
    attackInterval = {
        minimumValue = 0
    }
}
finalCore.placements = {
    {
        name = "normal",
        data = {
            health = 10,
            attackInterval = 2.0
        }
    }
}

function finalCore.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("characters/final_core/dormant", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 0.5)
    sprite.color = {0.5, 0.0, 0.5, 1.0}
    return sprite
end

function finalCore.selection(room, entity)
    return utils.rectangle(entity.x - 24, entity.y - 24, 48, 48)
end

return finalCore

File: Ahorn/entities/FinalCore.jl
module FinalCore

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/FinalCore" FinalCore(x::Integer, y::Integer, attackInterval::Number=2.0, health::Integer=10)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(FinalCore)
)

function Ahorn.selection(entity::FinalCore)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::FinalCore, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end
